ssl/ssl_locl.h

```diff
@@ -1132,6 +1136,12 @@ struct ssl_st {
     size_t psksession_id_len;
     /* Default generate session ID callback. */
     GEN_SESSION_CB generate_session_id;
+    /*
+     * The temporary TLSv1.3 session id. This isn't really a session id at all
+     * but is a random value sent in the legacy session id field.
+     */
+    unsigned char tmp_session_id[SSL_MAX_SSL_SESSION_ID_LENGTH];
+    size_t tmp_session_id_len;
```

Comment:
I actually did start down that route (i.e. reusing the existing session_id) but backed out the changes and went down this route instead. It started having all sorts of subtle consequences and caused breakage. Perhaps it could have been worked through, but in the end I decided that this wasn't actually a session id at all - it is used for a completely different purpose - so reusing the existing session_id field was probably the wrong way to go. It actually adds unnecessary complexity.

Reply:
I'll trust you that it doesn't work out very well, and agree that it is in some sense an object of a semantically different type.